docs/testplan/MACsec-test-plan.md

```diff
@@ -0,0 +1,445 @@
+   - The inject port expects to capture the encrypted PFC packet
+3. Send clear PFC frame on the PTF injected port
+   - The DUT expects to capture the clear PFC packet
+4. Send encrypted PFC frame on the PTF injected port
+   - The DUT expects to capture the clear PFC packet
+
+##### Strict mode
+
+The switch should only react encrypted PFC frames, send encrypted PFC frames.
+
+1. Send clear PFC frame from the neighbor device to the DUT
+   - The DUT expects to capture the clear PFC packet
+2. Send clear PFC frame from the DUT to the neighbor device
+   - The neighbor expect to capture the clear PFC packet
+   - The inject port expects to capture the encrypted PFC packet
+3. Send clear PFC frame on the PTF injected port
+   - The DUT expects to drop any PFC packet
+4. Send encrypted PFC frame on the PTF injected port
+   - The DUT expects to capture the clear PFC packet
+
+#### Verify SNMP request/response works across interface with macsec configuration
+
+- Configure the macsec profile on interface and check if the snmp walk succeeds from the peer VM.
+
+### Testcase : Deployment usecases and fault handling scenario's
+
+  This testcase covers the various fault scenario's and the expected behavior.
+
+#### Link flap on an interface with macsec configured
+
+- MKA session can be recovered from the link flap if the port comes back up in < 6 secs (MKA lietime)
+- If the port is down for more than 6 sec, MKA session will create a new session.
+
+#### Link flap on a portchannel member which has macsec configured
+
+- When the member interface flaps and it is the only portchannel member
+   - Expect the Portchannel to go down and come up depending on whether the member port comes back in 6 secs (MKA lietime)
+- When one member interface flaps, but the Portchannel has more member ports which are macsec enabled.
+   - Expect Portchannel to remain up.
+
+#### MACsec session cannot be established under wrong MKA configuration
+
+- If the CAK is mis-matched between DUT and peer, the MKA session cannot be established.
+   - Control plane protocols eg: BGP session will not be established.
+
+#### Config reload done on DUT with macsec configuration
+
+- The macsec sessions will be reconfigured, MKA session created again with a new SAK key.
+- The control protocol sessions like LACP, LLDP, BGP get established again over the macsec configured interfaces.
+
+#### Everflow, port mirroring on macsec enabled interfaces
+
+- This test is to verify 2 cases
+   - configure mirroring on interfaces where macsec is configured.
+   - configure macsec on the outgoing interface ( in case of Everflow ).
+     - TODO add expected behavior
+
+### Testcase : Macsec scenario's for multi-asic, multi-dut
+
+- Verify that macsec dockers are coming up in different namespaces.
+- Verify macsec packet flow where the Ingress and Egress ports are on different Linecards.
+   - TODO add expected behavior
+
+### Testcase : Scale tests
```

Comment:
Can power consumption test be added when macsec enabled?

Reply:
I'm not familiar about the power consumption test, how can we test it? is it possible to monitor this index?